lib/ansible/plugins/become/sudo.py

```diff
@@ -94,7 +96,20 @@ def build_become_command(self, cmd, shell):
         if self.get_option('become_pass'):
             self.prompt = '[sudo via ansible, key=%s] password:' % self._id
             if flags:  # this could be simplified, but kept as is for now for backwards string matching
-                flags = flags.replace('-n', '')
+                reflag = []
+                for flag in shlex.split(flags):
+                    if flag in ('-n', '--non-interactive'):
+                        continue
+                    elif not flag.startswith('--'):
+                        # handle -XnxxX flags only
+                        flag = re.sub(r'^(-\w*)n(\w*.*)', r'\1\2', flag)
+                    reflag.append(flag)
+                try:
+                    flags = shlex.join(reflag)
+                except AttributeError:
+                    # for older pythons
+                    flags = ' '.join([shlex.quote(a) for a in reflag])
```

Comment:
`shlex.quote` isn't available on Python 2.x.

Reply:
i could swear we were using that one for a long time ... I'll update